### PR TITLE
openzfs-dev 2.0.0 (new cask)

### PR DESCRIPTION
**Pull request by yurikoles**
Signed-off-by: Yurii Kolesnykov <root@yurikoles.com>

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [ ] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-versions/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew install --cask {{cask_file}}` worked successfully.
- [x] `brew uninstall --cask {{cask_file}}` worked successfully.

### CODE REVIEW

**Comment by yurikoles** [comment]
```
Casks/openzfs-dev.rb:4:3: C: Convert if-elsif to case-when.
```
But HBC fails to parse cask with case-when and I didn't see any examples in any tap I have. So I think that this check should be disabled.

**Comment by yurikoles** [comment]
Hi @miccal,

I'm sorry that I screwed your commits by force-push, I had tried to re-apply your changes and added you as co-author.

In the past I was asked to squash and force-push my changes in HBC, now it's no longer a case?

File: Casks/openzfs-dev.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  desc "ZFS driver and utilities"
+  homepage "https://openzfsonosx.org/"
+
+  conflicts_with cask: "openzfs"
+  depends_on macos: ">= :mojave"
+
+  pkg "OpenZFSonOsX.pkg"
+
+  preflight do
+    FileUtils.ln_sf(staged_path/"file.php", staged_path/"OpenZFSonOsX.pkg")
```
Comment (yurikoles) [inline Casks/openzfs-dev.rb:28]: For some unknown reason HBC stages `.pkg` artifact with the name `file.php` despite the fact that it uses the correct file name when downloading and cashing it.

**Comment by yurikoles** [comment]
Ping?

File: Casks/openzfs-dev.rb (hunk 1 above)
Comment (reitermarkus) [reply Casks/openzfs-dev.rb:24]: This should not be needed. Should be fixed in https://github.com/Homebrew/brew/pull/10085.

**Review by miccal — APPROVED** [review]

**Pull request merged**